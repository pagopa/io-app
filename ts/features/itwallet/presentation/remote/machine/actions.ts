import { useIONavigation } from "../../../../../navigation/params/AppParamsList.ts";
import { ITW_REMOTE_ROUTES } from "../navigation/routes.ts";
import ROUTES from "../../../../../navigation/routes.ts";
import { ITW_ROUTES } from "../../../navigation/routes.ts";

export const createRemoteActionsImplementation = (
  navigation: ReturnType<typeof useIONavigation>
) => ({
  navigateToFailureScreen: () => {
    navigation.navigate(ITW_REMOTE_ROUTES.MAIN, {
      screen: ITW_REMOTE_ROUTES.FAILURE
    });
  },

  navigateToDiscoveryScreen: () => {
    navigation.navigate(ITW_ROUTES.MAIN, {
      screen: ITW_ROUTES.DISCOVERY.INFO
    });
  },

  navigateToWallet: () => {
    navigation.reset({
      index: 1,
      routes: [
        {
          name: ROUTES.MAIN,
          params: {
            screen: ROUTES.WALLET_HOME
          }
        }
      ]
    });
  },

  closeIssuance: () => {
    navigation.popToTop();
  }
});
