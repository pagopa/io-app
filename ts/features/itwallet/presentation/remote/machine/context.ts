import { RequestObject } from "../../../common/utils/itwTypesUtils";
import {
  EnrichedPresentationDetails,
  ItwRemoteRequestPayload,
  RelyingPartyConfiguration
} from "../utils/itwRemoteTypeUtils";
import { RemoteFailure } from "./failure";

export type Context = {
  /**
   * The remote request payload for the remote presentation
   */
  payload: ItwRemoteRequestPayload | undefined;
  /**
   * The failure of the remote presentation machine
   */
  failure?: RemoteFailure;
  /**
   * Relying party Entity Configuration subject
   */
  rpSubject: string | undefined;
  /**
   * Relying party Entity Configuration metadata
   */
  rpConf: RelyingPartyConfiguration | undefined;
  /**
   * The Encoded Request Object fetched from the Relying Party in the authorization request
   */
  requestObjectEncodedJwt: string | undefined;
  /**
   * The Request Object fetched from the Relying Party with the presentation details
   */
  requestObject: RequestObject | undefined;
  /**
   * Details of the presentation requested by the Relying Party
   * It includes the requested claims and credentials
   */
  presentationDetails: EnrichedPresentationDetails | undefined;
  /**
   * Optional credentials selected by the user, identified by their presentation ID
   */
  selectedOptionalCredentials: Set<string>;
  /**
   * The URI to redirect the user to access the Relying Party's service
   * It is not required in cross-device presentation
   */
  redirectUri?: string;
};

export const InitialContext: Context = {
  payload: undefined,
  failure: undefined,
  rpSubject: undefined,
  rpConf: undefined,
  requestObjectEncodedJwt: undefined,
  requestObject: undefined,
  presentationDetails: undefined,
  selectedOptionalCredentials: new Set()
};
