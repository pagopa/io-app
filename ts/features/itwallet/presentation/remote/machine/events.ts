import { ItwRemoteRequestPayload } from "../Utils/itwRemoteTypeUtils.ts";

export type Reset = {
  type: "reset";
};

export type Start = {
  type: "start";
  payload: ItwRemoteRequestPayload;
};

export type GoToWalletActivation = {
  type: "go-to-wallet-activation";
};

export type GoToWallet = {
  type: "go-to-wallet";
};

export type Back = {
  type: "back";
};

export type Close = {
  type: "close";
};

export type RemoteEvents =
  | Reset
  | Start
  | GoToWalletActivation
  | GoToWallet
  | Back
  | Close;
