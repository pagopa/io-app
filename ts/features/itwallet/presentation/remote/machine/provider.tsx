import { createActorContext } from "@xstate/react";
import { itwRemoteMachine } from "./machine.ts";
import { createRemoteActorsImplementation } from "./actors.ts";
import { createRemoteActionsImplementation } from "./actions.ts";
import { createRemoteGuardsImplementation } from "./guards.ts";

type Props = {
  children: JSX.Element;
};

export const ItwRemoteMachineContext = createActorContext(itwRemoteMachine);

export const ItwRemoteMachineProvider = (props: Props) => {
  const remoteMachine = itwRemoteMachine.provide({
    guards: createRemoteGuardsImplementation(),
    actions: createRemoteActionsImplementation(),
    actors: createRemoteActorsImplementation()
  });

  return (
    <ItwRemoteMachineContext.Provider logic={remoteMachine}>
      {props.children}
    </ItwRemoteMachineContext.Provider>
  );
};
