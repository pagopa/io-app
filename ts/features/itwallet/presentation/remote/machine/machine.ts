import { assign, setup } from "xstate";
import { ItwTags } from "../../../machine/tags";
import { InitialContext, Context } from "./context";
import { mapEventToFailure } from "./failure";
import { RemoteEvents } from "./events";

export const itwRemoteMachine = setup({
  types: {
    context: {} as Context,
    events: {} as RemoteEvents
  },
  actions: {
    setFailure: assign(({ event }) => ({ failure: mapEventToFailure(event) }))
  },
  actors: {},
  guards: {}
}).createMachine({
  id: "itwRemoteMachine",
  context: { ...InitialContext },
  initial: "Idle",
  states: {
    Idle: {
      description:
        "The machine is in idle, ready to start the remote presentation flow",
      on: {
        start: {
          actions: assign(({ event }) => ({
            payload: event.payload
          })),
          target: "RemoteRequestValidation"
        }
      }
    },
    RemoteRequestValidation: {
      description: "Validating the remote request payload before proceeding",
      tags: [ItwTags.Loading],
      // WARNING: for mocking purposes only
      after: {
        1000: {
          target: "ClaimsDisclosure"
        }
      }
    },
    ClaimsDisclosure: {
      description:
        "Display the list of claims to disclose for the verifiable presentation"
    },
    Failure: {
      description: "This state is reached when an error occurs"
    }
  }
});

export type ItwRemoteMachine = typeof itwRemoteMachine;
