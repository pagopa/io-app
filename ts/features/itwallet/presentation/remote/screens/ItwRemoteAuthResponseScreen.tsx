import { Linking } from "react-native";
import { IOToast } from "@pagopa/io-app-design-system";
import { OperationResultScreenContent } from "../../../../../components/screens/OperationResultScreenContent";
import I18n from "../../../../../i18n";
import { ItwRemoteLoadingScreen } from "../components/ItwRemoteLoadingScreen";
import { ItwRemoteMachineContext } from "../machine/provider";
import {
  selectIsLoading,
  selectIsSuccess,
  selectRedirectUri,
  selectRelyingPartyData
} from "../machine/selectors";

export const ItwRemoteAuthResponseScreen = () => {
  const machineRef = ItwRemoteMachineContext.useActorRef();
  const isLoading = ItwRemoteMachineContext.useSelector(selectIsLoading);
  const isSuccess = ItwRemoteMachineContext.useSelector(selectIsSuccess);
  const rpData = ItwRemoteMachineContext.useSelector(selectRelyingPartyData);
  const redirectUri = ItwRemoteMachineContext.useSelector(selectRedirectUri);

  /**
   * In addition to checking for the loading state,
   * we need to ensure that the current state is not `Success`
   * to prevent a visual glitch caused by a slight delay in navigation
   */
  if (isLoading || !isSuccess) {
    return (
      <ItwRemoteLoadingScreen
        title={I18n.t(
          "features.itWallet.presentation.remote.loadingScreen.response",
          { relyingParty: rpData?.organization_name }
        )}
      />
    );
  }

  const closeMachine = () => machineRef.send({ type: "close" });

  return (
    <OperationResultScreenContent
      pictogram="success"
      title={I18n.t("features.itWallet.presentation.remote.success.title")}
      subtitle={I18n.t(
        "features.itWallet.presentation.remote.success.subtitle"
      )}
      action={
        redirectUri
          ? {
              icon: "externalLinkSmall",
              label: I18n.t(
                "features.itWallet.presentation.remote.success.cta"
              ),
              onPress: () => {
                Linking.openURL(redirectUri)
                  .catch(() => IOToast.error("global.genericError"))
                  .finally(closeMachine);
              }
            }
          : {
              label: I18n.t("global.buttons.close"),
              onPress: closeMachine
            }
      }
    />
  );
};
