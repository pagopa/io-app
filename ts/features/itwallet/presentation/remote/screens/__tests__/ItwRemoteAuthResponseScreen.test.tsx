import { createActor } from "xstate";
import { render } from "@testing-library/react-native";
import { itwRemoteMachine } from "../../machine/machine";
import { ItwRemoteMachineContext } from "../../machine/provider";
import { ItwRemoteAuthResponseScreen } from "../ItwRemoteAuthResponseScreen";

describe("ItwRemoteAuthResponseScreen", () => {
  it("should match snapshot when no redirect_uri is available", () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  it("should match snapshot when a redirect_uri is available", () => {
    expect(renderComponent("https://myrp/auth/success")).toMatchSnapshot();
  });
});

const renderComponent = (redirectUri?: string) => {
  const initialSnapshot = createActor(itwRemoteMachine).getSnapshot();
  const snapshot: typeof initialSnapshot = {
    ...initialSnapshot,
    value: "Success",
    context: {
      ...initialSnapshot.context,
      redirectUri
    }
  };

  return render(
    <ItwRemoteMachineContext.Provider options={{ snapshot }}>
      <ItwRemoteAuthResponseScreen />
    </ItwRemoteMachineContext.Provider>
  );
};
