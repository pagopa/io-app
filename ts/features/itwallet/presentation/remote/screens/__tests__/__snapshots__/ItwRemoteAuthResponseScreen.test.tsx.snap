// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItwRemoteAuthResponseScreen should match snapshot when a redirect_uri is available 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flexGrow": 1,
      "marginHorizontal": 24,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
      ]
    }
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      [
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={120}
          bbWidth={120}
          focusable={false}
          height={120}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 120,
                "width": 120,
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
          width={120}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="m113.74 163 1.1-8.41 5.3-3.55 5.06-.4 1.57-4.88.83-5.55 5.66-3.03 7.35-.79 5.07 1.07 4.32 5.48 2.47-19.83 3.92-8.98 5.78-5.32 4.53-.75c.29-2.62.44-5.29.44-7.99C167.13 59.71 134.42 27 94.07 27 53.72 27 21 59.71 21 100.07c0 40.36 32.71 73.07 73.07 73.07 7.71 0 15.14-1.2 22.12-3.42l-2.45-6.72Z"
              fill={
                {
                  "payload": 4289392367,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M142.56 202.43c-1.34 0-2.8-.15-4.38-.44-11.03-2.03-18.79-11.47-21.33-19.38-1.55-4.83-1.17-8.91 1.04-11.2l2.88 2.78c-1.1 1.15-1.15 3.97-.11 7.2 2.19 6.81 8.83 14.93 18.24 16.67 4.38.81 7.34.32 8.79-1.45 2.45-2.98.69-9.42-.19-11.65l3.72-1.48c.42 1.04 3.94 10.34-.43 15.67-1.79 2.19-4.56 3.29-8.24 3.29l.01-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M146.3 185.03c-3.04 0-6.72-.53-11.06-1.59-17.32-4.23-21.57-16.39-22.56-23.26-.54-3.75.43-7.46 2.67-10.18 2.92-3.55 6.82-4.24 10.97-1.93l-1.94 3.5c-2.44-1.36-4.27-1.06-5.94.97-1.52 1.85-2.18 4.43-1.8 7.07.85 5.87 4.51 16.27 19.55 19.94 10.44 2.55 15.19 1.37 16.16-.59.72-1.45-.4-3.66-1.49-4.54l2.51-3.11c2.54 2.05 4.19 6.14 2.56 9.42-1.41 2.85-4.64 4.28-9.63 4.28v.02Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M152.04 174.91c-4.89 0-11.5-3.06-20.94-9.63-5.64-3.93-8.3-10.28-7.68-18.38.57-7.5 6.31-13.6 13.35-14.17 2.09-.17 6.42-.52 9.73 2.42 2.47 2.19 3.81 5.75 3.99 10.57l-4 .15c-.14-3.69-1.03-6.29-2.65-7.73-1.88-1.67-4.47-1.61-6.75-1.42-5.1.41-9.26 4.92-9.68 10.48-.51 6.7 1.5 11.68 5.98 14.8 9.75 6.79 16.31 9.65 20.07 8.76 1.4-.33 2.45-1.21 3.29-2.76.81-1.49-.55-3.8-1.42-5.05-4.33-6.17-14.65-12.19-18.35-11.63l-.6-3.95c6.01-.92 17.61 6.71 22.22 13.28 3.12 4.44 2.63 7.49 1.67 9.26-1.38 2.55-3.36 4.15-5.88 4.75-.73.17-1.51.26-2.34.26l-.01-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M199.72 163.99c-7.16 0-14.29-1.78-21.16-5.33-6.79-3.51-11.4-10.13-12.34-17.7-.51-4.15.37-9.46 1.22-14.59 1.01-6.09 2.16-12.98.19-15.3-.33-.39-.94-.9-2.49-.9-2.91 0-5.32 1.04-7.39 3.19-7.96 8.27-7.27 29.08-7.26 29.29l-4 .15c-.03-.92-.75-22.73 8.37-32.21 2.82-2.93 6.27-4.41 10.27-4.41 2.38 0 4.24.78 5.54 2.31 3.16 3.73 1.97 10.92.7 18.54-.8 4.85-1.64 9.86-1.2 13.44.79 6.36 4.51 11.69 10.21 14.64 11.84 6.13 24.52 6.48 36.67 1.03l1.64 3.65c-6.25 2.8-12.63 4.2-18.98 4.2h.01ZM194.04 212.26c-6.74 0-14.48-.86-23.27-3.07-13.54-3.42-23.05-8.39-23.44-8.6l1.87-3.54c.37.2 37.68 19.53 67.92 6.96l1.54 3.69c-5.69 2.36-13.86 4.55-24.61 4.55l-.01.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M88.22 123.5c-.4 0-.79-.16-1.07-.45L61.92 97.31c-.58-.59-.57-1.54.02-2.12.59-.58 1.54-.57 2.12.02l24.13 24.62 42.36-45.48c.56-.61 1.52-.64 2.12-.08.61.56.64 1.51.08 2.12l-43.43 46.63c-.28.3-.67.47-1.08.48h-.02Z"
              fill={
                {
                  "payload": 4278240714,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "height": 24,
            }
          }
        />
      </View>
      <Text
        accessibilityRole="header"
        allowFontScaling={true}
        dynamicTypeRamp="title2"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#0E0F13",
              "fontFamily": "Titillio",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "600",
              "lineHeight": 33,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Fatto!
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <Text
        allowFontScaling={true}
        dynamicTypeRamp="body"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#555C70",
              "fontFamily": "Titillio",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Continua sul sito dell'ente.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Continua sul sito dell'ente"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "auto",
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderCurve": "continuous",
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "textAlignVertical": "center",
                  },
                  {
                    "paddingHorizontal": 24,
                  },
                  {
                    "backgroundColor": "#0B3EE3",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 8,
                    "borderWidth": 0,
                    "height": 48,
                  },
                  {},
                  false,
                  {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "columnGap": 8,
                    },
                    false,
                  ]
                }
              >
                <RNSVGSvgView
                  accessibilityElementsHidden={true}
                  accessibilityLabel=""
                  accessible={false}
                  align="xMidYMid"
                  animatedProps={
                    {
                      "color": undefined,
                    }
                  }
                  bbHeight={30}
                  bbWidth={30}
                  color="#555C70"
                  focusable={false}
                  height={30}
                  importantForAccessibility="no-hide-descendants"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {
                        "flex": 0,
                        "height": 30,
                        "width": 30,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                  width={30}
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGPath
                      clipRule={0}
                      d="M7 8C7 7.44772 7.44772 7 8 7H8.75C9.30228 7 9.75 6.55228 9.75 6C9.75 5.44772 9.30228 5 8.75 5H8C6.34315 5 5 6.34315 5 8V16C5 17.6569 6.34315 19 8 19H16C17.6569 19 19 17.6569 19 16V15.25C19 14.6977 18.5523 14.25 18 14.25C17.4477 14.25 17 14.6977 17 15.25V16C17 16.5523 16.5523 17 16 17H8C7.44772 17 7 16.5523 7 16V8ZM13 5C12.4477 5 12 5.44772 12 6C12 6.55228 12.4477 7 13 7H15.5858L12.2929 10.2929C11.9024 10.6834 11.9024 11.3166 12.2929 11.7071C12.6834 12.0976 13.3166 12.0976 13.7071 11.7071L17 8.41421V11C17 11.5523 17.4477 12 18 12C18.5523 12 19 11.5523 19 11V7C19 5.89543 18.1046 5 17 5H13Z"
                      fill={
                        {
                          "type": 2,
                        }
                      }
                      fillRule={0}
                      propList={
                        [
                          "fill",
                          "fillRule",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  accessibilityElementsHidden={true}
                  accessible={false}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  importantForAccessibility="no-hide-descendants"
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    [
                      {},
                      {
                        "color": "#0E0F13",
                        "fontFamily": "Titillio",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "600",
                        "lineHeight": undefined,
                      },
                      [
                        {
                          "textAlign": "auto",
                        },
                        {
                          "color": undefined,
                        },
                      ],
                    ]
                  }
                >
                  Continua sul sito dell'ente
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;

exports[`ItwRemoteAuthResponseScreen should match snapshot when no redirect_uri is available 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "flexGrow": 1,
      "marginHorizontal": 24,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    collapsable={false}
    contentContainerStyle={
      [
        {
          "alignContent": "center",
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        false,
      ]
    }
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    waitFor={
      [
        {
          "current": null,
        },
      ]
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={120}
          bbWidth={120}
          focusable={false}
          height={120}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 120,
                "width": 120,
              },
            ]
          }
          vbHeight={240}
          vbWidth={240}
          width={120}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          >
            <RNSVGPath
              d="m113.74 163 1.1-8.41 5.3-3.55 5.06-.4 1.57-4.88.83-5.55 5.66-3.03 7.35-.79 5.07 1.07 4.32 5.48 2.47-19.83 3.92-8.98 5.78-5.32 4.53-.75c.29-2.62.44-5.29.44-7.99C167.13 59.71 134.42 27 94.07 27 53.72 27 21 59.71 21 100.07c0 40.36 32.71 73.07 73.07 73.07 7.71 0 15.14-1.2 22.12-3.42l-2.45-6.72Z"
              fill={
                {
                  "payload": 4289392367,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M142.56 202.43c-1.34 0-2.8-.15-4.38-.44-11.03-2.03-18.79-11.47-21.33-19.38-1.55-4.83-1.17-8.91 1.04-11.2l2.88 2.78c-1.1 1.15-1.15 3.97-.11 7.2 2.19 6.81 8.83 14.93 18.24 16.67 4.38.81 7.34.32 8.79-1.45 2.45-2.98.69-9.42-.19-11.65l3.72-1.48c.42 1.04 3.94 10.34-.43 15.67-1.79 2.19-4.56 3.29-8.24 3.29l.01-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M146.3 185.03c-3.04 0-6.72-.53-11.06-1.59-17.32-4.23-21.57-16.39-22.56-23.26-.54-3.75.43-7.46 2.67-10.18 2.92-3.55 6.82-4.24 10.97-1.93l-1.94 3.5c-2.44-1.36-4.27-1.06-5.94.97-1.52 1.85-2.18 4.43-1.8 7.07.85 5.87 4.51 16.27 19.55 19.94 10.44 2.55 15.19 1.37 16.16-.59.72-1.45-.4-3.66-1.49-4.54l2.51-3.11c2.54 2.05 4.19 6.14 2.56 9.42-1.41 2.85-4.64 4.28-9.63 4.28v.02Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M152.04 174.91c-4.89 0-11.5-3.06-20.94-9.63-5.64-3.93-8.3-10.28-7.68-18.38.57-7.5 6.31-13.6 13.35-14.17 2.09-.17 6.42-.52 9.73 2.42 2.47 2.19 3.81 5.75 3.99 10.57l-4 .15c-.14-3.69-1.03-6.29-2.65-7.73-1.88-1.67-4.47-1.61-6.75-1.42-5.1.41-9.26 4.92-9.68 10.48-.51 6.7 1.5 11.68 5.98 14.8 9.75 6.79 16.31 9.65 20.07 8.76 1.4-.33 2.45-1.21 3.29-2.76.81-1.49-.55-3.8-1.42-5.05-4.33-6.17-14.65-12.19-18.35-11.63l-.6-3.95c6.01-.92 17.61 6.71 22.22 13.28 3.12 4.44 2.63 7.49 1.67 9.26-1.38 2.55-3.36 4.15-5.88 4.75-.73.17-1.51.26-2.34.26l-.01-.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M199.72 163.99c-7.16 0-14.29-1.78-21.16-5.33-6.79-3.51-11.4-10.13-12.34-17.7-.51-4.15.37-9.46 1.22-14.59 1.01-6.09 2.16-12.98.19-15.3-.33-.39-.94-.9-2.49-.9-2.91 0-5.32 1.04-7.39 3.19-7.96 8.27-7.27 29.08-7.26 29.29l-4 .15c-.03-.92-.75-22.73 8.37-32.21 2.82-2.93 6.27-4.41 10.27-4.41 2.38 0 4.24.78 5.54 2.31 3.16 3.73 1.97 10.92.7 18.54-.8 4.85-1.64 9.86-1.2 13.44.79 6.36 4.51 11.69 10.21 14.64 11.84 6.13 24.52 6.48 36.67 1.03l1.64 3.65c-6.25 2.8-12.63 4.2-18.98 4.2h.01ZM194.04 212.26c-6.74 0-14.48-.86-23.27-3.07-13.54-3.42-23.05-8.39-23.44-8.6l1.87-3.54c.37.2 37.68 19.53 67.92 6.96l1.54 3.69c-5.69 2.36-13.86 4.55-24.61 4.55l-.01.01Z"
              fill={
                {
                  "payload": 4278927075,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M88.22 123.5c-.4 0-.79-.16-1.07-.45L61.92 97.31c-.58-.59-.57-1.54.02-2.12.59-.58 1.54-.57 2.12.02l24.13 24.62 42.36-45.48c.56-.61 1.52-.64 2.12-.08.61.56.64 1.51.08 2.12l-43.43 46.63c-.28.3-.67.47-1.08.48h-.02Z"
              fill={
                {
                  "payload": 4278240714,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <View
          style={
            {
              "height": 24,
            }
          }
        />
      </View>
      <Text
        accessibilityRole="header"
        allowFontScaling={true}
        dynamicTypeRamp="title2"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#0E0F13",
              "fontFamily": "Titillio",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "600",
              "lineHeight": 33,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Fatto!
      </Text>
      <View
        style={
          {
            "height": 8,
          }
        }
      />
      <Text
        allowFontScaling={true}
        dynamicTypeRamp="body"
        maxFontSizeMultiplier={1.5}
        style={
          [
            {},
            {
              "color": "#555C70",
              "fontFamily": "Titillio",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "400",
              "lineHeight": 24,
            },
            {
              "textAlign": "center",
            },
          ]
        }
      >
        Continua sul sito dell'ente.
      </Text>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <View
          style={
            {
              "height": 24,
            }
          }
        />
        <View>
          <View
            accessibilityLabel="Close"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignSelf": "auto",
                "flexShrink": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "borderCurve": "continuous",
                    "elevation": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "overflow": "hidden",
                    "textAlignVertical": "center",
                  },
                  {
                    "paddingHorizontal": 24,
                  },
                  {
                    "backgroundColor": "#0B3EE3",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 8,
                    "borderWidth": 0,
                    "height": 48,
                  },
                  {},
                  false,
                  {
                    "backgroundColor": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    {
                      "columnGap": 8,
                    },
                    false,
                  ]
                }
              >
                <Text
                  accessibilityElementsHidden={true}
                  accessible={false}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  importantForAccessibility="no-hide-descendants"
                  maxFontSizeMultiplier={1.5}
                  numberOfLines={1}
                  style={
                    [
                      {},
                      {
                        "color": "#0E0F13",
                        "fontFamily": "Titillio",
                        "fontSize": 16,
                        "fontStyle": "normal",
                        "fontWeight": "600",
                        "lineHeight": undefined,
                      },
                      [
                        {
                          "textAlign": "auto",
                        },
                        {
                          "color": undefined,
                        },
                      ],
                    ]
                  }
                >
                  Close
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RNCSafeAreaView>
`;
